### ISSUE TITLE
Forbid constant as condition in `if`

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1005]
# Rule request

## Thesis

Bad:

```python
if 1: ...
if True: ...
if False: ...
if None: ...
if '': ...
```

We already have a rule forbidding magic numbers. Also, Pylint checks the unreachable code. It should cover a few cases above, but only a few.

Good:

```python
TYPING = False

if TYPING:
   ...

while True: ...
```

## Reasoning

I should say something here, but I'm not sure about the possible reasoning behind it. Is it typo? Fast commenting out the code for debugging?

**Comment by sobolevn** [comment]
It is already supported.

```python
  1:1      WPS314 Conditional always evaluates to same result
  if 1: ...
  ^

  2:1      WPS314 Conditional always evaluates to same result
  if True: ...
  ^

  3:1      WPS314 Conditional always evaluates to same result
  if False: ...
  ^

  4:1      WPS314 Conditional always evaluates to same result
  if None: ...
  ^

  5:1      WPS314 Conditional always evaluates to same result
  if '': ...
```
